src/thor/isochrone_action.cc

```diff
@@ -37,6 +37,10 @@ std::string thor_worker_t::isochrones(Api& request) {
                                                                         : ExpansionType::forward;
   auto grid = isochrone_gen.Expand(expansion_type, request, *reader, mode_costing, mode);
 
+  // e.g. in case of /expansion request
+  if (only_expand)
+    return "";
```

Comment:
i see why you wanted to pass in the flag, because you can do so from the expansion action in the worker, the root cause of having to do this is that we dont store the intermediate information of the result of the isochrone in the protobuf. if we did we would return from here having modified the protobuf and then just send the protobuf to a serialization function separately. if that were the case we wouldnt have to return a blank string here we would just return after the expansion and then the serializer would call generate contours and do the serialization of those.

its not worth the refactoring in my opinion to do this now but i think to fit the pattern we have elsewhere and avoid any trickery it might be worth doing the refactoring at some point so every action works the same, basically: look in the protobuf to see what to do, add the work you did to the protobuf and return.

Reply:
> we dont store the intermediate information of the result of the isochrone in the protobuf

yep would be much nicer..

this was also where I wasn't sure which one you prefer:

1. pass in the flag to the isochrone_action (advantage: other (future) callers of the action could ask for `only_expand`)
2. inspect the `options.action()` right here and return if it's `expansion` ("only" advantage: cleaner/more consistent in thor actions ctors)

I prefer 1. but can understand if you want 2. and eventually refactor to not need this at all as you mentioned.

Reply:
I went with 2.